Breeding-Insight/bi-web
# [BI-1347] - Germplasm pedigree view

# Description
**Story:** [BI-1347 -  Germplasm Pedigree View](https://breedinginsight.atlassian.net/browse/BI-1347)

Integrated the BrAPI Pedigree Viewer into the germplasm view.

Implementation notes:
- made forks of both the pedigree viewer and brapi-js libraries from the solgenomics repo in order to expand functionality and make some bug fixes.
- the build GitHub Action was updated to use node 12, and then update npm to v8.  This was because of the use of HTTP links for the `brapi-js`, `brapi-pedigree-viewer`, and `d3-pedigree-tree` dependencies

**NOTE**: Before merging this PR:
- the BrAPI-Pedigree-Viewer dependency needs to be updated to point to the merge commit of https://github.com/Breeding-Insight/BrAPI-Pedigree-Viewer/pull/2.
- the BrAPI-JS dependency needs to be updated to point to the merge commit of https://github.com/Breeding-Insight/BrAPI-js/pull/1.


Out of scope:
- Utilize configuration in PedigreeViewer to allow pedigree to link to corresponding node's germplasm details page - This will be handled in [BI-1494](https://breedinginsight.atlassian.net/browse/BI-1494)

# Dependencies
- https://github.com/Breeding-Insight/BrAPI-Pedigree-Viewer/pull/2
- https://github.com/Breeding-Insight/BrAPI-js/pull/1

# Testing
Has been tested during development with a program pointed to Breedbase only (hasn't been tested on the BrAPI Java Server)

1. Load germplasm with multiple pedigree connections into your program
2. Navigate to a germplasm that has pedigree connections
3. Verify that the pedigree tree renders, and you can click on any of the arrows (up, down, right) to expand the pedigree tree
4. Verify that when your mouse is over the tree you can zoom in and out by scrolling your mouse
5. Verify that you can move the tree by clicking and dragging

# Checklist:

- [x] I have run TAF: https://github.com/Breeding-Insight/taf/runs/6814332383?check_suite_focus=true -> results match daily run of: https://github.com/Breeding-Insight/taf/runs/6806342693?check_suite_focus=true
